pkgs/development/libraries/libpng/default.nix

```
@@ -22,6 +22,12 @@ in stdenv.mkDerivation (finalAttrs: {
   };
   postPatch = whenPatched "gunzip < ${patch_src} | patch -Np1";
 
+  # feenableexcept is locked behind BSD_VISIBLE on FreeBSD, but enabling it
+  # messes up basically everything else. Just punt on implicit declarations.
+  env.CFLAGS = lib.optionalString stdenv.hostPlatform.isFreeBSD ''
+    -Wno-implicit-function-declaration
+  '';
+
```

can avoid mass rebuilds by using `configureFlags` and then this change can go into master.

```suggestion
  configureFlags = lib.optionals stdenv.hostPlatform.isFreeBSD [
    # feenableexcept is locked behind BSD_VISIBLE on FreeBSD, but enabling it
    # messes up basically everything else. Just punt on implicit declarations.
    "CFLAGS=-Wno-implicit-function-declaration"
  ];

```

---

_crosses fingers that I'm doing the rebase right_

---

Looks like that worked :) Let's see what ofborg has to say!

---

nice work.

alternatively:

```diff
diff --git a/pkgs/development/libraries/libpng/default.nix b/pkgs/development/libraries/libpng/default.nix
index 4a8c52ae279d..11e3fb31604f 100644
--- a/pkgs/development/libraries/libpng/default.nix
+++ b/pkgs/development/libraries/libpng/default.nix
@@ -20,7 +20,11 @@ in stdenv.mkDerivation (finalAttrs: {
     url = "mirror://sourceforge/libpng/libpng-${finalAttrs.version}.tar.xz";
     hash = "sha256-alygZSOSotfJ2yrltAIQhDwLvAgcvUEIJasAzFnxSmw=";
   };
-  postPatch = whenPatched "gunzip < ${patch_src} | patch -Np1";
+  postPatch = whenPatched "gunzip < ${patch_src} | patch -Np1"
+  + lib.optionalString stdenv.isFreeBSD ''
+
+    sed -i 1i'int feenableexcept(int __mask);' contrib/libtests/pngvalid.c
+  '';
 
   outputs = [ "out" "dev" "man" ];
   outputBin = "dev";
```

works too with no rebuilds (the extra \n before sed is required as to not end up an argument to patch). tho perhaps too nitpicky. either way works for me.